src/components/ha-icon-picker.ts

```diff
@@ -0,0 +1,165 @@
+import "@polymer/paper-input/paper-input";
+import { mdiCheck } from "@mdi/js";
+import { css, html, LitElement, TemplateResult } from "lit";
+import { ComboBoxLitRenderer, comboBoxRenderer } from "lit-vaadin-helpers";
+import { customElement, property, query } from "lit/decorators";
+import { fireEvent } from "../common/dom/fire_event";
+import { PolymerChangedEvent } from "../polymer-types";
+import "./ha-icon";
+import { mdiDeprecatedIcons } from "../data/iconsets";
+
+let cachedState: string[];
+
+// eslint-disable-next-line lit/prefer-static-styles
+const rowRenderer: ComboBoxLitRenderer<string> = (item) => html`<style>
+    paper-icon-item {
+      padding: 0;
+      margin: -8px;
+    }
+    #content {
+      display: flex;
+      align-items: center;
+    }
+    ha-svg-icon {
+      padding-left: 2px;
+      color: var(--secondary-text-color);
+    }
+    :host(:not([selected])) ha-svg-icon {
+      display: none;
+    }
+    :host([selected]) paper-icon-item {
+      margin-left: 0;
+    }
+  </style>
+
+  <ha-svg-icon .path=${mdiCheck}></ha-svg-icon>
+  <paper-icon-item>
+    <ha-icon .icon=${item} slot="item-icon"></ha-icon>
+    <paper-item-body> ${item} </paper-item-body>
+  </paper-icon-item>`;
+
+@customElement("ha-icon-picker")
+export class HaIconPicker extends LitElement {
+  @property() public value?: string;
+
+  @property() public label?: string;
+
+  @property() public placeholder?: string;
+
+  @property({ attribute: "error-message" }) public errorMessage?: string;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  @query("vaadin-combo-box-light", true) private comboBox!: HTMLElement;
+
+  private _initedStates = false;
+
+  private _states: string[] = [];
+
+  @property({ type: Boolean }) private _opened = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <vaadin-combo-box-light
```

Comment:
Use `ha-combo-box` instead, so all the input logic is taken care of.
```suggestion
      <ha-combo-box
```